docker/build.gradle

```diff
@@ -0,0 +1,553 @@
+/**
```

Comment:
@conradsp this appears to be a verbatim copy/paste of the `build.gradle` at the root of the project? We should avoid this - you know how these things go, I guarantee that this won't be kept up to date... if the reason that you to do this was because you couldn't get Docker to access files outside of the `docker/` sub-directory you're introducing here, then if I were you I would probably just put the `Dockerfile` at the root - not the end of the world, IMHO.